docs/presigned_urls.md

```diff
@@ -0,0 +1,354 @@
+
+The usage of the configuration file creation script via command line is as follows:
+
+```
+python3 generate_config_files.py --ids IDS --bucket BUCKET [--cohort COHORT] --cancer CANCER --assembly ASSEMBLY --sv SV --cnv CNV [--drivers DRIVERS] [--snv SNV] [--indel INDEL] [--reads READS] [--configs CONFIGS] [--expiration EXPIR]
+```
+Required parameters provide information needed to access the S3 bucket (`bucket`), identify the samples that will be added to the configuration file (`ids`), and define the required properties within the configuration file (`cancer`, `assembly`, `sv`, and `cnv`). The optional parameters provide additional information for optional properties within a configuration file (`drivers`, `snv`, `indel`, and `reads`), and details for the generated presigned URLs (`configs` and `expiration`). Additional details can be found in the table below:
+
+| Parameter | Default | Description | Corresponding Configuration File Property |
+|---|---|---|---|
+| `ids` | - | Required. The name of the sample ID TSV file within S3 bucket. | `id` |
+| `bucket` | - | Required. The name of the S3 bucket containing the private data objects. | - |
+| `cohort` | `None` | Optional. The name of the cohort subdirectory within the S3 bucket. | - |
+| `cancer` | - | Required. The type of cancer of the samples. | `cancer` |
+| `assembly` | `'hg38'` or `'hg19'` | Required. The reference genome assembly for the samples. | `assembly` |
+| `sv` | - | Required. The name of the subdirectory within the S3 bucket containing structural variant (SV) `bedpe` files. | `sv` |
+| `cnv` | - | Required. The name of the subdirectory within the S3 bucket containing copy number variant (CNV) `tsv` files. | `cnv` |
+| `drivers` | `None` | Optional. The name of the subdirectory within the S3 bucket containing driver mutation `tsv` or `json` files. | `drivers` |
+| `snv` | `None` | Optional. The name of the subdirectory within the S3 bucket containing point mutation (SNV) `vcf` files and their corresponding index `tbi` files. | `vcf`, `vcfIndex` |
+| `indel` | `None` | Optional. The name of the subdirectory within the S3 bucket containing indel `vcf` files and their corresponding index `tbi` files.  | `vcf2`, `vcf2Index` |
+| `reads` | `None` | Optional. The name of the subdirectory within the S3 bucket containing read alignment `bam` files and their corresponding index `bai` files. | `bam`, `bamIndex` |
+| `configs` | `'CONFIGS'` | Optional. The name of the subdirectory, both locally and within the S3 bucket, that will contain the generated Chromoscope configuration `json` file. | - |
+| `expiration` | `3600` | Optional. The duration of the presigned URLs within and for the generated Chromoscope configuration file, in seconds. | - |
+
+?> The `note` property of the configuration file is not handled here, since it is a direct textual annotation. Only properties that require a URL value (in addition to required properties) are defined using this script.
```

Comment:
could load the note from the file. if it were a command line arg, it would be the same note for all samples, which is probably  not what most users would want -- for now I am still going to omit this and leave this note (for time's sake), and let developers modify the script themselves if they would like to